modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/ComplimentaryDate.jsx

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import ClayDatePicker from '@clayui/date-picker';
+import {ClayCheckbox} from '@clayui/form';
+import {useState} from 'react';
+import {Link} from 'react-router-dom';
+import i18n from '../../../../../common/I18n';
+import {Button} from '../../../../../common/components';
+import Layout from '../../../../../common/containers/setup-forms/Layout';
+
+const ComplimentaryDate = ({
+	infoSelectedKey,
+	setInfoSelectedKey,
+	setStep,
+	urlPreviousPage,
+}) => {
+	const [selectedSubscription] = useState(
+		infoSelectedKey?.selectedSubscription
+	);
+	const [expandedOnOrAfter, setExpandedOnOrAfter] = useState(false);
+	const [selectedStartDate, setSelectedStartDate] = useState('');
+	const [checkBoxConfirmationTerms, setCheckBoxConfirmationTerms] = useState(
+		false
+	);
+
+	const now = new Date();
+	const NAVIGATION_YEARS_RANGE = 2;
+
+	const inputStartDate = new Date(selectedStartDate);
+	const timestamp = inputStartDate.getTime();
+	const timezoneOffset = inputStartDate.getTimezoneOffset() * 60000;
+	const startDateFormatted = new Date(timestamp + timezoneOffset);
+	const startDate = new Date(timestamp + timezoneOffset);
+	const endDate = new Date(
+		startDateFormatted.setDate(startDateFormatted.getDate() + 30)
+	);
```

Comment:
@eduardo-diniz I'm kinda a worried about those variables here, since you are using dates, the way its, will cause rerenders and recalculation for this dates.

Could you wrap it into a useMemo, and make this an object?

something like:

```js
const {inputStartDate, timestamp, timestampOffset, ...othervalueshere} = useMemo(() => ({
    inputStartDate: new Date(selectedStartDate),
    timestamp: inputStartDate.getTime(), 
    ... other values here....
}
), [probablySomeVariableHere])
```

Reply:
Thanks Master Kevin!

useMemo is beautiful.